test/common/deployment_types.go

```diff
@@ -183,6 +189,28 @@ func getDeploymentConfiguration(deploymentName string, inst *integreatlyv1alpha1
 		},
 	}
 
+	if inst.Spec.Type == string(integreatlyv1alpha1.InstallationTypeManagedApi) {
+		ratelimitCR := &v12.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      quota.RateLimitName,
+				Namespace: Marin3rProductNamespace,
+			},
+		}
+
+		key, err := k8sclient.ObjectKeyFromObject(ratelimitCR)
+		if err != nil {
+			t.Fatalf("Error getting key from ratelimit Deployment: %v", err)
+		}
+
+		err = ctx.Client.Get(context.TODO(), key, ratelimitCR)
+		if err != nil {
+			if !k8sError.IsNotFound(err) {
+				t.Fatalf("Error obtaining ratelimit CR: %v", err)
+			}
+		}
+		deployment["marin3rDeployment"].Products[1].ExpectedReplicas = *ratelimitCR.Spec.Replicas
```

Comment:
```suggestion
		deployment["marin3rDeployment"] = Namespace{
			Name: Marin3rProductNamespace,
			Products: []Product{
				{Name: "prom-statsd-exporter", ExpectedReplicas: 1},
				{Name: "ratelimit", ExpectedReplicas: *ratelimitCR.Spec.Replicas},
			},
		}
```
What do you think on just adding the `deployment["marin3rDeployment"]` block from https://github.com/integr8ly/integreatly-operator/blob/3ae07fc47fd73ee86a7e18c86054b8d2f40c709b/test/common/deployment_types.go#L149-L155 to  here instead of just changing the replica value ?  I feel it would be slighly safer on the off chance someone in the future adds to this and messes up the array order causing the wrong struct replica value to be reassigned. Also this will make it a single place where it is added and not changed somewhere else in the code :thinking:

Reply:
Yes, that's definitely the change to be made. Thanks